contracts/pcv/aave/AavePCVDeposit.sol

```diff
@@ -0,0 +1,93 @@
+}
+
+interface IncentivesController {
+    function claimRewards(address[] calldata assets, uint256 amount, address to) external;
+
+    function getRewardsBalance(address[] calldata assets, address user) external view returns(uint256);
+}
+
+/// @title Aave PCV Deposit
+/// @author Fei Protocol
+contract AavePCVDeposit is WethPCVDeposit {
+
+    /// @notice the associated Aave aToken for the deposit
+    IERC20 public aToken;
+
+    /// @notice the Aave v2 lending pool
+    LendingPool public lendingPool;
+
+    /// @notice the underlying token of the PCV deposit
+    IERC20 public token;
+
+    /// @notice the Aave incentives controller for the aToken
+    IncentivesController public incentivesController;
+
+    /// @notice Aave PCV Deposit constructor
+    /// @param _core Fei Core for reference
+    /// @param _lendingPool the Aave v2 lending pool
+    /// @param _token the underlying token of the PCV deposit
+    /// @param _aToken the associated Aave aToken for the deposit
+    /// @param _incentivesController the Aave incentives controller for the aToken
+    constructor(
+        address _core,
+        LendingPool _lendingPool,
+        IERC20 _token,
+        IERC20 _aToken,
+        IncentivesController _incentivesController
+    ) CoreRef(_core) {
+        lendingPool = _lendingPool;
+        aToken = _aToken;
+        token = _token;
+        incentivesController = _incentivesController;
+    }
+
+    /// @notice claims Aave rewards from the deposit and transfers to this address
+    function claimRewards() external {
+        address[] memory assets = new address[](1);
+        assets[0] = address(aToken);
+        // First grab the available balance
+        uint256 amount = incentivesController.getRewardsBalance(assets, address(this));
+
+        // claim all available rewards
+        incentivesController.claimRewards(assets, amount, address(this));
+    }
+
+    /// @notice deposit buffered aTokens
+    function deposit() external override whenNotPaused {
+        // wrap any held ETH if present
+        wrapETH();
+
+        // Approve and deposit buffered tokens
+        uint256 pendingBalance = token.balanceOf(address(this));
+        token.approve(address(lendingPool), pendingBalance);
+        lendingPool.deposit(address(token), pendingBalance, address(this), 0);
+        
+        emit Deposit(msg.sender, pendingBalance);
+    }
+
+    /// @notice withdraw tokens from the PCV allocation
+    /// @param amountUnderlying of tokens withdrawn
+    /// @param to the address to send PCV to
+    function withdraw(address to, uint256 amountUnderlying)
+        external
+        override
+        onlyPCVController
+    {
+        lendingPool.withdraw(address(token), amountUnderlying, to);
+        emit Withdrawal(msg.sender, to, amountUnderlying);
+    }
+
+    /// @notice returns total balance of PCV in the Deposit
+    function balance() public view override returns (uint256) {
+        return aToken.balanceOf(address(this));
```

Comment:
is that the balance of underlying tokens, or the amount of aTokens ?

Reply:
nvm :

![image](https://user-images.githubusercontent.com/6916865/129461295-dd8982df-5627-49ca-931c-b40783e91a04.png)

Reply:
Good call, shouldn't expect devs to know aTokens are rebasing